packages/apollo-server-express/src/ApolloServer.ts

```diff
@@ -142,12 +169,49 @@ export class ApolloServer extends ApolloServerBase {
     // schema, you'll need to manually specify `introspection: true` in the
     // ApolloServer constructor; by default, the introspection query is only
     // enabled in dev.
-    const guiEnabled =
-      !!gui || (gui === undefined && process.env.NODE_ENV !== 'production');
-
     app.use(path, (req, res, next) => {
-      if (guiEnabled && req.method === 'GET') {
+      // We cannot do an existence check on `gui` since it may be a boolean
+      const nonDynamicGui: NonDynamicGuiConfig =
+        gui !== undefined && gui !== null
+          ? isNonDynamic(gui)
+            ? gui
+            : gui(req)
+          : {};
+      const enabled: boolean = isBoolean(nonDynamicGui) ? nonDynamicGui : true;
+      const partialGuiOverrides: Partial<GuiOptions> = isBoolean(nonDynamicGui)
+        ? {}
+        : nonDynamicGui;
+
+      // Disallow endpoints in pre-fill settings
```

Comment:
I don't think we actually need to do this. I think this was my misunderstanding what "endpoint" is

Reply:
You're assertion is correct. The endpoint is where that tab is pointed at